File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = [0, 0]
    
    while s != "1":
        n = len(s) - s.count("0")
        answer[1] = answer[1] + s.count("0")
        answer[0] = answer[0] + 1
        new_s = ""
        while n > 0:
            new_s += str(n % 2)
            n = n // 2
                
        s = new_s
        
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let target = [...s]
    var answer = [0, 0];
    while(true) {
        if (target.length === 1) break
        answer[0]++
        const filteredTarget = target.filter(el => el !== "0")
        answer[1] += target.length - filteredTarget.length
        target = [...filteredTarget.length.toString(2)]
    }
    
    return answer;
}